From 92473f8169118472f0f40179eb60cb2cde765bb7 Mon Sep 17 00:00:00 2001
From: Ahmad Fatoum <a.fatoum@pengutronix.de>
Date: Sat, 23 Apr 2022 12:24:34 +0200
Subject: [PATCH 4/5] ARM: cpu: board-dt-2nd: call arm_cpu_lowlevel_init

The generic DT image could be started by boot firmware that doesn't do
all the initialization that we do in arm_cpu_lowlevel_init(), so call it
always for good measure. This enables using the generic image as second
stage to the Raspberry Pi videocore.

Signed-off-by: Ahmad Fatoum <a.fatoum@pengutronix.de>
---
 arch/arm/cpu/board-dt-2nd.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm/cpu/board-dt-2nd.c b/arch/arm/cpu/board-dt-2nd.c
index 0731aecd1a51..6f4a6f26a8c7 100644
--- a/arch/arm/cpu/board-dt-2nd.c
+++ b/arch/arm/cpu/board-dt-2nd.c
@@ -21,6 +21,8 @@ void dt_2nd_aarch64(void *fdt)
 
 	/* entry point already set up stack */
 
+	arm_cpu_lowlevel_init();
+
 	relocate_to_current_adr();
 	setup_c();
 
@@ -50,6 +52,8 @@ ENTRY_FUNCTION(start_dt_2nd, r0, r1, r2)
 {
 	unsigned long image_start = (unsigned long)_text + global_variable_offset();
 
+	arm_cpu_lowlevel_init();
+
 	arm_setup_stack(image_start);
 
 	relocate_to_current_adr();
-- 
2.30.2

